pkg/loadtest/local_load_test.go

```diff
@@ -28,145 +28,105 @@ import (
-			}
-
-			blockAdapter := testutil.NewBlockAdapterByType(t, blockstoreType)
-			c, err := catalog.New(ctx, catalog.Config{
-				Config:  conf,
-				KVStore: storeMessage,
-			})
-			testutil.MustDo(t, "build catalog", err)
-
-			source := catalog.NewActionsSource(c)
-			outputWriter := catalog.NewActionsOutputWriter(c.BlockAdapter)
-
-			// wire actions
-			actionsService := tt.actionsService(t, ctx, source, outputWriter, &stats.NullCollector{}, true)
-			c.SetHooksHandler(actionsService)
-
-			credentials, err := auth.SetupAdminUser(ctx, tt.authService, superuser)
-			testutil.Must(t, err)
-
-			authenticator := auth.NewBuiltinAuthenticator(tt.authService)
-			kvParams, err := conf.GetKVParams()
-			testutil.Must(t, err)
-			migrator := kv.NewDatabaseMigrator(kvParams)
-			t.Cleanup(func() {
-				_ = c.Close()
-			})
-			auditChecker := version.NewDefaultAuditChecker(conf.GetSecurityAuditCheckURL(), "")
-			emailParams, _ := conf.GetEmailParams()
-			emailer, err := email.NewEmailer(emailParams)
-			testutil.Must(t, err)
-			handler := api.Serve(
-				conf,
-				c,
-				authenticator,
-				authenticator,
-				tt.authService,
-				blockAdapter,
-				tt.meta,
-				migrator,
-				&stats.NullCollector{},
-				nil,
-				actionsService,
-				auditChecker,
-				logging.Default(),
-				emailer,
-				nil,
-				nil,
-				nil,
-				nil,
-				nil,
-				upload.DefaultPathProvider,
-			)
-
-			ts := httptest.NewServer(handler)
-			defer ts.Close()
-
-			testConfig := Config{
-				FreqPerSecond:    6,
-				Duration:         10 * time.Second,
-				MaxWorkers:       math.MaxInt64,
-				KeepRepo:         false,
-				Credentials:      *credentials,
-				ServerAddress:    ts.URL,
-				StorageNamespace: "mem://local/test/",
-			}
-			loader := NewLoader(testConfig)
-			err = loader.Run()
-			if err != nil {
-				t.Fatalf("Got error on test: %s", err)
-			}
-		})
+	kvStore := kvtest.GetStore(ctx, t)
+	storeMessage := &kv.StoreMessage{Store: kvStore}
+	authService := auth.NewKVAuthService(storeMessage, crypt.NewSecretStore([]byte("some secret")), nil, authparams.ServiceCache{}, logging.Default().WithField("service", "auth"))
+	meta := auth.NewKVMetadataManager("local_load_test", conf.GetFixedInstallationID(), conf.GetDatabaseType(), kvStore)
+
+	blockstoreType := os.Getenv(testutil.EnvKeyUseBlockAdapter)
+	if blockstoreType == "" {
+		blockstoreType = "mem"
+	}
+
+	storeMessage = &kv.StoreMessage{Store: kvStore}
+	viper.Set("database.kv_enabled", true)
+
+	blockAdapter := testutil.NewBlockAdapterByType(t, blockstoreType)
+	c, err := catalog.New(ctx, catalog.Config{
+		Config:  conf,
+		KVStore: storeMessage,
+	})
+	testutil.MustDo(t, "build catalog", err)
+
+	source := catalog.NewActionsSource(c)
+	outputWriter := catalog.NewActionsOutputWriter(c.BlockAdapter)
+
+	// wire actions
+	actionsService := actions.NewService(ctx, actions.NewActionsKVStore(kv.StoreMessage{Store: kvStore}), source, outputWriter, &actions.DecreasingIDGenerator{}, &stats.NullCollector{}, true)
```

Comment:
You already have to storeMessage

Reply:
Need to update the actions to use pointer or interface instead of a copy